routers/repo/issue_timetrack.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+	"net/http"
+	"strconv"
+)
+
+// AddTimeManual tracks time manually
+func AddTimeManual(c *context.Context) {
+	hours, err := strconv.ParseInt(c.Req.PostForm.Get("hours"), 10, 64)
+	if err != nil {
+		hours = 0
```

Comment:
`strconv.ParseInt` allready default to 0 (https://golang.org/pkg/strconv/#ParseInt), it would be better to fail if errors. We could check if hours is empty and use 0 in this case but if someone give letter to hours we should fail.